overlord/servicestate/quota_control.go

```diff
@@ -201,6 +201,12 @@ func UpdateQuota(st *state.State, name string, updateOpts QuotaGroupUpdate) (*st
 		return nil, fmt.Errorf("cannot update group %q: %v", name, err)
 	}
 
+	// ensure that the group we are modifying does not contain a mix of snaps and sub-groups
+	// as we no longer support this, and existing quota groups might have this
+	if err := ensureGroupIsNotMixed(name, allGrps); err != nil {
+		return nil, err
```

Comment:
this case does not seem to be triggered by tests

Reply:
This is now fixed, had not seen that this was not covered the by the quota_handlers, added a similar test in quota_control. Thanks for this!